src/dummy_daos/daos.h

```diff
@@ -16,12 +16,12 @@
 
 #pragma once
 
-#include <stdlib.h>
-#include <stdint.h>
+#include <cstdlib>
```

Comment:
The file is written as a C header, i.e. it features a `ifdef __cplusplus`, changing the c headers to c++ headers will break the c compatibility. 

This file probably does not need to expose a C-API, but in that case the Ifdefs should also be removed as they give the impression of C-compatiblity